net_infrastructure/comware_5_2_port.py

```diff
@@ -0,0 +1,265 @@
+                      or is invalid!")
+
+        if self._port_changed(facts, port):
+            facts = self._save_port(facts, port)
+        if self.module.params.get('save') is True:
+            self.save()
+
+        return facts
+
+    def _port_changed(self, facts, port):
+        tagged = 'tagged'
+        if port['tagged'] is False:
+            tagged = "un" + tagged
+
+        # TODO: solve this
+        # current_port = facts['current_config']['interfaces'][port['name']]
+        # TODO: 'port link_type' needs to be 'link-type'
+        #if port['port link-type'] != current_port['link_type']:
+        #    return True
+        #if port['vlans'] != current_port['vlan'][tagged][port['link_type']]:
+        #    return True
+
+        return True
+
+    def _save_port(self, facts, port):
+        # if something has changed, best to delete then recreate
+        self.set_changed(False)
+        # could use else, but this is self-documenting
+        tagged = 'tagged'
+        if not port['tagged']:
+            tagged = 'un' + tagged
+
+        if port['link_type'] == 'access' and port['tagged']:
+            self.fail("A link-type of 'access' cannot be tagged")
+        if port['link_type'] == 'access' and len(port['vlans']) > 1:
+            self.fail("A link-type of 'access' can only specify one vlan")
+
+        #current_link_type = \
+        # facts['current_config']['interfaces'][port['name']]['link_type']
+
+        self._send_command("interface %s\n" % port['name'],
+                           "ERROR: unable to enter interface view")
+        #if current_link_type != port['link_type'] and \
+        #        port['link_type'] != 'access':
+        if True:
+            self._send_command("port link-type acccess\n",
+                               "ERROR: unable to set link type to access")
+        type_err = "ERROR: unable to set link-type %s" % port['link_type']
+        self._send_command("port link-type %s\n" % port['link_type'], type_err)
+
+        vlan_list = " ".join(port['vlans'])
+        type_err = "Error: unable to set vlan %s access on port" % vlan_list
+        if port['link_type'] == 'hybrid':
+            self._send_command("port hybrid vlan %s %s\n" %
+                               (vlan_list, tagged), type_err)
+        elif port['link_type'] == 'trunk':
+            self._send_command("port trunk permit vlan %s\n" % vlan_list,
+                               type_err)
+        # access
+        else:
+            self._send_command("port access vlan %s\n" % vlan_list, type_err)
+
+        # leave interface view
+        self._quit()
+
+        # refresh facts
+        facts = self.get_facts()
+        self.set_changed(True)
+        self.append_message("PORT %s saved\n" % port['name'])
+
+        return facts
+
+
+def main():
+    module = AnsibleModule(
+        argument_spec=dict(
+            developer_mode=dict(type='bool'),
+            gather_facts=dict(required=False, type='bool', default=True),
```

Comment:
undocumented option, also we can probably move to 1st module to also gather port facts there.